tests/multi_ptr/multi_ptr_comparison_op.h

```diff
@@ -0,0 +1,586 @@
+            .with("address_space", address_space_name)
+            .with("decorated", is_decorated_name)
+            .create()) {
+      // Variable that contains all variables that will be used to verify test
+      // result
+      detail::mptr_mptr_test_results<T> expected_results;
+      // Expected value that will be returned after "first_mptr <= first_mptr"
+      // operator will be called
+      expected_results.first_first_mptr_result_val = true;
+      // Expected value that will be returned after "first_mptr <= second_mptr"
+      // operator will be called
+      expected_results.first_second_mptr_result_val = true;
+      // Expected value that will be returned after "second_mptr <= first_mptr"
+      // operator will be called
+      expected_results.second_first_mptr_result_val = false;
+
+      const auto run_test_action = [](auto arr_acc, auto result_acc) {
+        multi_ptr_t arr_mptr(arr_acc);
+        // multi_ptr that pointed at the first element
+        multi_ptr_t mptr_1(arr_mptr);
+        // multi_ptr that pointed at the second element
+        multi_ptr_t mptr_2(arr_mptr + 1);
+
+        auto &test_result = result_acc[0];
+
+        test_result.first_first_mptr_result_val = mptr_1 <= mptr_1;
+        test_result.first_second_mptr_result_val = mptr_1 <= mptr_2;
+        test_result.second_first_mptr_result_val = mptr_2 <= mptr_1;
+      };
+
+      run_test(queue, run_test_action, expected_results);
+    }
+    SECTION(
+        section_name(
+            "Check multi_ptr operator>=(const multi_ptr&, const multi_ptr&)")
+            .with("T", type_name)
+            .with("address_space", address_space_name)
+            .with("decorated", is_decorated_name)
+            .create()) {
+      // Variable that contains all variables that will be used to verify test
+      // result
+      detail::mptr_mptr_test_results<T> expected_results;
+      // Expected value that will be returned after "first_mptr >= first_mptr"
+      // operator will be called
+      expected_results.first_first_mptr_result_val = true;
+      // Expected value that will be returned after "second_mptr >= first_mptr"
+      // operator will be called
+      expected_results.first_second_mptr_result_val = true;
+      // Expected value that will be returned after "first_mptr >= second_mptr"
+      // operator will be called
+      expected_results.second_first_mptr_result_val = false;
+
+      const auto run_test_action = [](auto arr_acc, auto result_acc) {
+        multi_ptr_t arr_mptr(arr_acc);
+        // multi_ptr that pointed at the first element
+        multi_ptr_t mptr_1(arr_mptr);
+        // multi_ptr that pointed at the second element
+        multi_ptr_t mptr_2(arr_mptr + 1);
+
+        auto &test_result = result_acc[0];
+
+        test_result.first_first_mptr_result_val = mptr_1 >= mptr_1;
+        test_result.first_second_mptr_result_val = mptr_2 >= mptr_1;
+        test_result.second_first_mptr_result_val = mptr_1 >= mptr_2;
```

Comment:
There is a lot of copy-paste in all these tests.
At the end it looks like you could just generalize with
```c++
run_multi_ptr_test(..., some expected results, [](auto a, auto b) { return a >= b; },...);
```

Reply:
I can use fold expression with some lambdas and then iterate over all of them and call. We should explicitly write all lambdas that will be used. 
So, if we should write all lambdas this solution is not better than the current one